strategy/inuse.go

```diff
@@ -110,6 +116,43 @@ func (strat *inUseStragey) Run(policy v1alpha1.SnapshotPolicy) (*v1alpha1.Snapsh
 	return retStatus, retErr
 }
 
+func (strat *inUseStragey) getPvcNames(policy v1alpha1.SnapshotPolicy) ([]string, error) {
+	pvcNames := make(map[string]bool)
+	if policy.Spec.PVCNames != nil {
+		for _, pvcName := range *policy.Spec.PVCNames {
+			pvcNames[pvcName] = true
+		}
+	}
+	errors := make([]string, 0)
+	if policy.Spec.PVCLabelSelectors != nil {
+		for _, s := range *policy.Spec.PVCLabelSelectors {
+			selector, err := metav1.LabelSelectorAsSelector(&s)
+			if err != nil {
+				errors = append(errors, fmt.Sprintf("Selector %v: %v", s, err))
+				continue
+			}
+			pvcs, err := strat.pvcLister.PersistentVolumeClaims(policy.Namespace).List(selector)
+			if err != nil {
+				errors = append(errors, fmt.Sprintf("Selector %v: %v", selector, err))
+				continue
+			}
+			for _, pvc := range pvcs {
+				pvcNames[pvc.Name] = true
+			}
+		}
+	}
+
+	pvcArray := make([]string, 0, len(pvcNames))
```

Comment:
I actually wanted to pre-set the capacity so no reallocations are done because I already know the length. But also wanted to leverage the `append` so I don't have to keep an index variable in the loop that follows

https://github.com/cisco-sso/snapshotpolicy/blob/0454c0c868d1cd489a2995ac1f46909cbbd49bb2/strategy/inuse.go#L146-L148

Reply:
hm, any idea why is github not embedding the code and only highlighting it as a link?

Reply:
you're correct